internal/transport/http2_server.go

```diff
@@ -171,15 +171,10 @@ func NewServerTransport(conn net.Conn, config *ServerConfig) (_ ServerTransport,
 		ID:  http2.SettingMaxFrameSize,
 		Val: http2MaxFrameLen,
 	}}
-	// TODO(zhaoq): Have a better way to signal "no limit" because 0 is
-	// permitted in the HTTP2 spec.
-	maxStreams := config.MaxStreams
-	if maxStreams == 0 {
-		maxStreams = math.MaxUint32
-	} else {
+	if config.MaxStreams != math.MaxUint32 {
```

Comment:
Does it mean that it will NOT take effect any more if applications set MaxConcurrentStreams as `math.MaxUint32`?

Note the default value set by etcd is `math.MaxUint32`, reference:
- https://github.com/etcd-io/etcd/blob/a404ee5f6b8f0e3590b9abcc540d8813f09872aa/server/embed/serve.go#L264
- https://github.com/etcd-io/etcd/blob/a404ee5f6b8f0e3590b9abcc540d8813f09872aa/server/embed/config.go#L66

Reply:
It seems OK, since the [default maxConcurrentStreams is math.MaxUint32](https://github.com/grpc/grpc-go/blob/32e3ef1ed1c494e3d748b93d67a0b7ec5d089a49/server.go#L176)